src/arrayMethodJoin.js

```diff
@@ -5,7 +5,47 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```

Comment:
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```
Додай дефолтне значення для сепаратора, тоді можна буде видалити половину коду (там де else if)